### PR TITLE
Math Magicians: Components & Logic testing

### PR DESCRIPTION
**Pull request by lily4178993**
# Math Magicians: Components & Logic testing
This PR is raised in order to merge all units tests for all components and logic functions according to the project [requirements](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md)

## Changes implemented

- Write unit tests for the files `operate.js` and `calculate.js` using Jest.
- Use Jest snapshots to test the components.
- Use React Testing Library to simulate user interaction.

## How should this app be tested?
Clone this repository:
```sh
git clone https://github.com/lily4178993/Math-Magicians.git
```

Checkout the branch:
```sh
cd Math-Magicians
git checkout calc-tests
```

Install all dependencies:
```sh
npm install
```
Fix JS/JSX linter error
```sh
npx eslint "**/*.{js,jsx}" --fix
```
Open the project:
```sh
npm test
```

Please kindly review our implementation and suggest any changes and improvements.  Best regards.
Thanks for reviewing this PR 😊

### CODE REVIEW

**Review by ridwanediallo — COMMENTED** [review]
Hi @lily4178993 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🚩 🚩 🚩

**You submitted a project that you completed entirely on your own. However, because this is a pair-programming project, please ensure that your partner has contributed to it .**:ok_hand: 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: src/components/__tests__/Header.test.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { BrowserRouter as Router } from 'react-router-dom';
+import Header from 'components/Header';
+
+test('renders logo and navigation links correctly', () => {
+  render(
+    <Router>
```
Comment (TadesseDev) [inline src/components/__tests__/Header.test.js:8]: - [ ] You have done an impressive job team. But as the requirement states, there should be the usage of Jest snapshots for testing all the components. currently, you only have a snapshot test for the `Calculator` component. Please add a Jest snapshot test for the rest of the components.

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Most of the functionality has been tested. ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/__tests__/Header.test.js (hunk 1 above)
Comment (lily4178993) [reply src/components/__tests__/Header.test.js:8]: Thank you @TadesseDev.
Thank you for taking the time to review this pull request.
I took your requests into account and added Jest snapshot for the rest of the components.

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well to add snapshots for all the files as requested by the previous reviewer

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by lily4178993** [comment]
Thanks everyone for taking the time to review this branch.
Have a nice day.

**Pull request merged**